config/jobs/README.md

```diff
@@ -1,9 +1,146 @@
+    - For kubernetes/kubernetes we prefer `kubernetes/sig-foo/filename.yaml`
+    - Ensure `filename.yaml` is unique across the config subdir; prow uses this as a key in its configmap
+1. Ensure an [`OWNERS`](https://go.k8s.io/owners) file exists in the directory for job, and has appropriate approvers/reviewers
+1. Write or edit the job config (please see [how-to-add-new-jobs](/prow/jobs.md#how-to-configure-new-jobs))
+1. Ensure the job is configured to to display its results in [testgrid.k8s.io]
+    - The new way: add [testgrid annotations]
+    - The old way: update [`testgrid/config.yaml`]
+    - Please see the [testgrid README](/testgrid/README.md) for more details on configuation options
+1. Ensure any necessary formatting or config generation happens
+    - Run [`hack/update-config.sh`]
+    - CI will fail against your PR if this was necessary
+1. Open a PR with the changes; when it merges [@k8s-ci-robot] will deploy the changes automatically
+
+## Deleting Jobs
+
+1. Find the prowjob config file in this directory 
+1. Remove the entry for your job; if that was the last job in the file, remove the file
+1. If the job had no [testgrid annotations], ensure its [`testgrid/config.yaml`] entries are gone
+1. Ensure any necessary formatting or config generation happens
+    - Run [`hack/update-config.sh`]
+    - CI will fail against your PR if this was necessary
+1. Open a PR with the changes; when it merges [@k8s-ci-robot] will deploy the changes automatically
+
+## Testing Jobs Locally
+
+Please try using [`phaino`](/prow/cmd/phaino/README.md), it will interactively
+help you run a docker command that approximates the pod that would be scheduled
+on behalf of an actual prow job
+
+## Generated Jobs
+
+There are some sets of jobs that are generated and should not be edited by hand.
+These specific instructions should probably just live adjacent to the jobs rather
+than in this central file
+
+### kubernetes-security jobs
+
+These are automatically generated by [`hack/update-config.sh`] and should require no human intervention
+
+### image-validation jobs
+
+These test different master/node image versions against multiple k8s branches. If you
+want to change these, update [`experiment/test_config.yaml`](/experiment/test_config.yaml)
+and then run
+
+```
+bazel run //experiment:generate_tests -- --yaml-config-path=experiment/test_config.yaml
+```
+
+### release-branch jobs
+
+When a release branch of kubernetes is first cut, the current set of master jobs
+must be forked to use the new release branch. Use [`experiment/config-forker`] to
+accomplish this, eg:
+
+```
+# from test-infra root
+bazel run //experiment/config-forker -- \
+  --job-config $(pwd)/config/jobs \
+  --version 1.15 \
+  --output $(pwd)/config/jobs/kubernetes/sig-release/release-branch-jobs/1.15.yaml
+```
+
+[prow.k8s.io]: https://prow.k8s.io
+[@k8s-ci-robot]: https://github.com/k8s-ci-robot
+[testgrid annotations]: /testgrid/README.md#minimal-configuration
```

Comment:
You know how I said that this _probably_ wouldn't conflict with #13104

I was wrong; that refactor changes this header to "prow-job-configuration"

Reply:
I'm going to wait for that PR to merge before updating